### PR TITLE
docs(configuration): update output.library

### PR DESCRIPTION
**Pull request by chenxsan**
Related issue https://github.com/webpack/webpack.js.org/issues/4687

Preview url https://webpack-js-org-git-fork-chenxsan-feature-output-library-365426.vercel.app/configuration/output/#outputlibrary

### CODE REVIEW

File: src/content/configuration/output.mdx [hunk 1]
```diff
@@ -1305,6 +1305,206 @@ W> Note that not setting a `output.library` will cause all properties returned b
...
+  ```javascript
+  module.exports = {
+    //...
+    output: {
+      library: {
+        root: 'MyLibrary',
+        amd: 'my-library',
+        commonjs: 'my-common-library',
+      },
+      libraryTarget: 'umd',
```
Comment (alexander-akait) [inline src/content/configuration/output.mdx]: It can be rewritten on `library.name = { root, amd, commonjs }`, we should avoid using `libraryTarget`

Comment (alexander-akait) [reply src/content/configuration/output.mdx]: `libraryTarget` only for compatibility, all library options should be in `output.library`

Comment (chenxsan) [reply src/content/configuration/output.mdx]: All values of `libraryTarget` will have respective `library.type` item. For example, here's the new one for `umd` type https://github.com/webpack/webpack.js.org/pull/4691/files#diff-24a08fc6495329508b1a44d5f20bdaf472555d1c495157ee021f9405dea9bf5dR1056.

But I think we could still keep documentation for `libraryTarget` for a while so users can migrate. I'm going to inject a warning at the beginning of every `libraryTarget` encouraging users to use the respective `library.type`.

Comment (alexander-akait) [reply src/content/configuration/output.mdx]: Yep, just think we should provide all examples using `library`, and provide notes about `libraryTarget` with small example

**Comment by chenxsan** [comment]
Will update the [author library guide](https://webpack.js.org/guides/author-libraries/) in another pull request with links to the new `output.library.type` once this get merged.

**Comment by alexander-akait** [comment]
> Will update the author library guide in another pull request with links to the new output.library.type once this get merged.

Sounds good

**Review by alexander-akait — APPROVED** [review]
Good job

**Pull request merged**